src/mozilla_taskgraph/transforms/replicate.py

```diff
@@ -0,0 +1,248 @@
+                    """
+                Define which tasks to target for replication.
+
+                Each item in the list can be either:
+
+                    1. A taskId
+                    2. An index path that points to a single task
+
+                If any of the resolved tasks are a Decision task, targeted
+                tasks will be derived from the `task-graph.json` artifact.
+                """.lstrip()
+                ),
+            ): [str],
+            Optional(
+                "include-attrs",
+                description=dedent(
+                    """
+                A dict of attribute key/value pairs that targeted tasks will be
+                filtered on. Targeted tasks must *match all* of the given
+                attributes or they will be ignored.
+
+                Matching is performed by the :func:`~taskgraph.util.attrmatch`
+                utility function.
+                """.lstrip(),
+                ),
+            ): {str: Any(str, [str])},
+            Optional(
+                "exclude-attrs",
+                description=dedent(
+                    """
+                A dict of attribute key/value pairs that targeted tasks will be
+                filtered on. Targeted tasks must *not match any* of the given
+                attributes or they will be ignored.
+
+                Matching is performed by the :func:`~taskgraph.util.attrmatch`
+                utility function.
+                """.lstrip(),
+                ),
+            ): {str: Any(str, [str])},
+            Optional(
+                "include-deps",
+                description=dedent(
+                    """
+                If provided, dependencies of selected tasks will have their
+                upstream dependencies recursively walked to find additional tasks
+                to rerun in the staging instance. Any tasks matching one of the
+                given regex patterns will be rerun in the staging instance.
+                """.lstrip(),
+                ),
+            ): [str],
+        },
+    },
+    extra=ALLOW_EXTRA,
+)
+
+TASK_ID_RE = re.compile(
+    r"^[A-Za-z0-9_-]{8}[Q-T][A-Za-z0-9_-][CGKOSWaeimquy26-][A-Za-z0-9_-]{10}[AQgw]$"
+)
+
+transforms = TransformSequence()
+transforms.add_validate(REPLICATE_SCHEMA)
+
+
+@transforms.add
+def resolve_targets(config, tasks):
+    for task in tasks:
+        config = task.pop("replicate")
+
+        task_defs = []
+        for target in config["target"]:
+            if TASK_ID_RE.match(target):
+                # target is a task id
+                task_id = target
+            else:
+                # target is an index path
+                task_id = find_task_id(target)
+
+            try:
+                # we have a decision task, add all tasks from task-graph.json
+                result = get_artifact(task_id, "public/task-graph.json").values()
+                task_defs.extend(result)
+            except Exception:
```

Comment:
It would probably be better to be more targeted in this `except` block (or to check for the existence of the artifact rather than using `except` for flow control). We're likely to eventually eat real exceptions as written currently.